snapshot/git/gitdb/backends.go

```diff
@@ -0,0 +1,62 @@
+package gitdb
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/scootdev/scoot/snapshot"
+)
+
+type backend interface {
+	parseID(id string, kind string, parts []string) (snap, error)
+}
+
+type uploader interface {
+	upload(id snapshot.ID) (snapshot.ID, error)
+	backend
+}
+
+// GitDB uses several different Backends to store Snapshots.
+// snap is the common interface for a Snapshot (identified by an ID)
+type snap interface {
+	ID() snapshot.ID
+
+	Kind() snapKind
+	SHA() string
+	Download(db *DB) error
+}
+
+// parseID parses ID into a snap
+func (db *DB) parseID(id snapshot.ID) (snap, error) {
+	if id == "" {
+		return nil, fmt.Errorf("empty snapshot ID")
+	}
+	parts := strings.Split(string(id), "-")
+	if len(parts) < 3 {
+		return nil, fmt.Errorf("could not determine backend and kind in ID %s", id)
+	}
+
+	backendType, kind := parts[0], snapKind(parts[1])
+
+	if !kinds[kind] {
+		return nil, fmt.Errorf("invalid kind: %s", kind)
+	}
+
+	switch backendType {
+	case localIDText:
+		return db.local.parseID(id, kind, parts[2:])
+	case streamIDText:
+		return db.stream.parseID(id, kind, parts[2:])
+	default:
```

Comment:
Eh, I'd rather not commit to a plan in code here; let's just implement it over the next short period?